src/compiler/utilities.ts

```diff
@@ -5804,8 +5804,8 @@ function escapeTemplateSubstitution(str: string): string {
 /** @internal */
 export function hasInvalidEscape(template: TemplateLiteral): boolean {
     return template && !!(isNoSubstitutionTemplateLiteral(template)
-        ? template.templateFlags
-        : (template.head.templateFlags || some(template.templateSpans, span => !!span.literal.templateFlags)));
+        ? !!((template.templateFlags || 0) & TokenFlags.ContainsInvalidEscape)

```

Comment:
This seems super noisy; can we pull this repeated bit out into a function?

Reply:
ye, i had such a plan - pushed it out now 👍